docs/upcoming_changes/template.rst

```diff
@@ -0,0 +1,39 @@
+{% set title = "Numba {} Release Notes".format(versiondata.version) %}
+{{ "=" * title|length }}
+{{ title }}
+{{ "=" * title|length }}
+
+{% for section, _ in sections.items() %}
+{% set underline = underlines[0] %}{% if section %}{{ section }}
+{{ underline * section|length }}{% set underline = underlines[1] %}
+
+{% endif %}
+{% if sections[section] %}
+{% for category, val in definitions.items() if category in sections[section] %}
```

Comment:
We should change the ordering of the sections. Currently, the CUDA session is going before the new_features. I found this the ordering is driven by the sections ordering in towncrier.toml. I suggest the following changes to it:

```

[[tool.towncrier.type]]
    directory = "highlight"
    name = "Highlights"
    showcontent = true

[[tool.towncrier.type]]
    directory = "new_feature"
    name = "New Features"
    showcontent = true

[[tool.towncrier.type]]
    directory = "improvement"
    name = "Improvements"
    showcontent = true

[[tool.towncrier.type]]
    directory = "np_support"
    name = "NumPy Support"
    showcontent = true

[[tool.towncrier.type]]
    directory = "cuda"
    name = "CUDA API Changes"
    showcontent = true

[[tool.towncrier.type]]
    directory = "performance"
    name = "Performance Improvements and Changes"
    showcontent = true

[[tool.towncrier.type]]
    directory = "bug_fix"
    name = "Bug Fixes"
    showcontent = true

[[tool.towncrier.type]]
    directory = "change"
    name = "Changes"
    showcontent = true

[[tool.towncrier.type]]
    directory = "deprecation"
    name = "Deprecations"
    showcontent = true

[[tool.towncrier.type]]
    directory = "expired"
    name = "Expired Deprecations"
    showcontent = true

[[tool.towncrier.type]]
    directory = "compatibility"
    name = "Compatibility Notes"
    showcontent = true

[[tool.towncrier.type]]
    directory = "doc"
    name = "Documentation Changes"
    showcontent = true

[[tool.towncrier.type]]
    directory = "infrastructure"
    name = "Infrastructure Related Changes"
    showcontent = true
```